controller_manager:
  ros__parameters:
    update_rate: 100
    use_sime_time: true

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
    
    position_controller:
      type: position_controllers/JointGroupPositionController
    
    velocity_controller:
      type: velocity_controllers/JointGroupVelocityController
    
    effort_controller:
      type: effort_controllers/JointGroupEffortController
    
    joint_trajectory_controller:
      type: "joint_trajectory_controller/JointTrajectoryController"


joint_state_broadcaster:
  ros__parameters:
    joints:
      - joint_1
      - joint_2
      - joint_3
      - joint_4
      - joint_5
      - joint_6


position_controller:
  ros__parameters:
    joints:
      - joint_1
      - joint_2
      - joint_3
      - joint_4
      - joint_5
      - joint_6


velocity_controller:
  ros__parameters:
    joints:
      - joint_1
      - joint_2
      - joint_3
      - joint_4
      - joint_5
      - joint_6


effort_controller:
  ros__parameters:
    joints:
      - joint_1
      - joint_2
      - joint_3
      - joint_4
      - joint_5
      - joint_6


joint_trajectory_controller:
  ros__parameters:
    joints:
      - joint_1
      - joint_2
      - joint_3
      - joint_4
      - joint_5
      - joint_6

    command_interfaces:
      - position
      
    state_interfaces:
      - position
      - velocity

    state_publish_rate: 100.0
    action_monitor_rate: 20.0

    allow_partial_joints_goal: false
    open_loop_control: true
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.0
      joint_1:
        trajectory: 0.05
        goal: 0.03
  